addon/mixins/route.js

```diff
@@ -266,139 +181,73 @@ const RouteMixin = Mixin.create({
-      })
-      .finally(() => {
-        this.set('_loadingMore', false);
-      });
-  },
+  _loadNextPage(infinityModel) {
+    set(infinityModel, '_loadingMore', true);
 
-  /**
-   request the next page from the adapter
+    const modelName = get(infinityModel, '_infinityModelName');
+    const params    = infinityModel.buildParams();
 
-   @private
-   @method _requestNextPage
-   @returns {Ember.RSVP.Promise} A Promise that resolves the next page of objects
-   */
-  _requestNextPage() {
-    const modelName   = this.get('_infinityModelName');
-    const nextPage    = this.incrementProperty('currentPage');
-    const params      = this._buildParams(nextPage);
+    return this._requestNextPage(modelName, params)
+      .then(newObjects => this._afterInfinityModel(this)(newObjects, infinityModel))
+      .then(newObjects => this._doUpdate(newObjects, infinityModel))
+      .then(infinityModel => {
+        infinityModel.incrementProperty('currentPage');
+        set(infinityModel, '_firstPageLoaded', true);
 
-    return this.get(this._store)[this._storeFindMethod](modelName, params).then(
-      this._afterInfinityModel(this));
+        const canLoadMore = get(infinityModel, '_canLoadMore');
+        set(infinityModel, 'reachedInfinity', !canLoadMore);
+        if (!canLoadMore) { this._notifyInfinityModelLoaded(); }
+
+        return infinityModel;
+      }).finally(() => set(infinityModel, '_loadingMore', false));
   },
 
   /**
-   build the params for the next page request
+   request the next page from the adapter
 
    @private
-   @method _buildParams
-   @param {Number} nextPage the page number for the current request
-   @return {Object} The query params for the next page of results
+   @method _requestNextPage
+   @param {String} modelName
+   @param {Object} options
+   @returns {Ember.RSVP.Promise} A Promise that resolves the next page of objects
    */
-  _buildParams(nextPage) {
-    const pageParams = {};
-
-    if(this.get('perPageParam')){
-      pageParams[this.get('perPageParam')] = this.get('_perPage');
-    }
-
-    if(this.get('pageParam')){
-      pageParams[this.get('pageParam')] = nextPage;
-    }
-
-    const params = objectAssign(pageParams, this.get('_extraParams'));
-
-    const boundParams = this.get('_boundParams');
-    if (!Ember.isEmpty(boundParams)) {
-      Object.keys(boundParams).forEach(k => params[k] = this.get(boundParams[k]));
-    }
-
-    return params;
+  _requestNextPage(modelName, params) {
+    return this.get(this._store)[this._storeFindMethod](modelName, params);
   },
 
   /**
+   set _totalPages param on infinityModel
    Update the infinity model with new objects
-   Only called on the second page and following
-
-   @deprecated
-   @method updateInfinityModel
-   @param {Ember.Enumerable} newObjects The new objects to add to the model
-   @return {Ember.Array} returns the new objects
-   */
-  updateInfinityModel(newObjects) {
-    return this._doUpdate(newObjects);
-  },
```

Comment:
I'm pretty sure there's apps out there that override this method (it's public, after all) for the sake of old-school array proxies (where u pushed into content, rather than with content, or something)

Should we deprecate this, and provide a way to override this on the model?